evalml/pipelines/components/estimators/regressors/arima_regressor.py

```diff
@@ -142,7 +142,25 @@ def _match_indices(self, X, y):
     def _set_forecast(self, X):
         from sktime.forecasting.base import ForecastingHorizon
 
-        fh_ = ForecastingHorizon([i + 1 for i in range(len(X))], is_relative=True)
+        # we can only calculate the difference if the indices are of the same type
+        if isinstance(X.index[0], type(self.last_X_index)):
+            units_diff = X.index[0] - self.last_X_index
+            if isinstance(X.index, pd.DatetimeIndex):
+                dates_diff = pd.date_range(
+                    start=self.last_X_index,
+                    end=X.index[0],
+                    freq=X.index.freq,
+                )
+                units_diff = len(dates_diff) - 1
+            fh_ = ForecastingHorizon(
+                [units_diff + i for i in range(len(X))],
```

Comment:
Seems like we needed an off-by-one offset in the non-gap case. Should that still be the case here as well?

Reply:
We need the + 1 since `range(len(X))` starts at 0. In the gap case `units_diff` should always be > 0 so we don't need it!